src/layer_norm.cpp

```diff
@@ -0,0 +1,135 @@
+                                ConstData_t bias,
+                                const TensorDescriptor& yDesc,
+                                Data_t y,
+                                const TensorDescriptor& meanDesc,
+                                Data_t mean,
+                                const TensorDescriptor& rstdDesc,
+                                Data_t rstd,
+                                miopenLayerNormMode_t mode,
+                                const float epsilon,
+                                const int32_t normalized_dim)
+{
+    if(x == nullptr || y == nullptr)
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Null pointer for tensor.");
+    }
+
+    if(xDesc.GetType() != yDesc.GetType())
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Tensor types do not match.");
+    }
+
+    if(xDesc.GetLengths() != yDesc.GetLengths())
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Tensor dimension lengths do not match.");
+    }
+
+    bool is_all_packed = xDesc.IsPacked() && weightDesc.IsPacked() && biasDesc.IsPacked() &&
+                         yDesc.IsPacked() && meanDesc.IsPacked() && rstdDesc.IsPacked();
+
+    if(!is_all_packed)
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "All tensor is not packed.");
+    }
+
+    auto dims         = xDesc.GetLengths();
+    size_t grid_size  = 1;
+    size_t outer_size = 1;
+    size_t inner_size = 1;
+    size_t i          = 0;
+    for(; i < normalized_dim; i++)
+    {
+        outer_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    for(; i < dims.size(); i++)
+    {
+        inner_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    auto dtype = xDesc.GetType();
+
+    const std::vector<size_t> vld{LOCAL_SIZE, 1, 1};
+    const std::vector<size_t> vgd{outer_size * vld[0], 1, 1};
+
+    std::string algo_name = "LayerNormForward";
+    std::string network_config =
+        "lnfwd-dtype" + std::to_string(static_cast<int32_t>(dtype)) + "g" + std::to_string(vgd[0]) +
+        "l" + std::to_string(vld[0]) + "normalized_dim" + std::to_string(normalized_dim) + "grid" +
+        std::to_string(grid_size) + "outer_size" + std::to_string(outer_size) + "inner_size" +
+        std::to_string(inner_size) + "mode" + std::to_string(static_cast<int32_t>(mode)) + "eps" +
+        std::to_string(static_cast<float>(epsilon));
+
+    std::string program_name = "MIOpenLayerNorm.cpp";
+    std::string kernel_name  = "LayernormFwdContiguous";
+
+    // compile parameters
+    std::string parms =
+        " -DMIOPEN_USE_FP16=" + std::to_string(static_cast<int32_t>(dtype == miopenHalf)) +
+        " -DMIOPEN_USE_FP32=" + std::to_string(static_cast<int32_t>(dtype == miopenFloat)) +
+        " -DMIOPEN_USE_FP64=" + std::to_string(static_cast<int32_t>(dtype == miopenDouble)) +
+        " -DMIOPEN_USE_BFP16=" + std::to_string(static_cast<int32_t>(dtype == miopenBFloat16));
+
+    parms += " -DMIOPEN_BETA_API=1";
+    parms += " -DLOCAL_SIZE=" + std::to_string(LOCAL_SIZE);
```

Comment:
We do not use this old-style machinery in new solvers.

Reply:
If you have any examples about new-style, please let me know.

Reply:
Agree, it would be nice to have some examples.

Reply:
@CAHEK7 @seungmanhan Please look how the `KernelBuildParameters` is used in the library.